bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,338 @@
+
+The plugin attempts to force the provisioning of the deployment scanner in dev mode:
+
+* If deployment-scanner is excluded, remove it from exclusion. A warning message is displayed advertising that the provisioning of the deployment-scanner is enforced.
+* Add the deployment-scanner layer to the set of layers. An info message is displayed advertising that the deployment-scanner is provisioned.
+* If a provisioning.xml file is set (and no layers override it), a warning is displayed advertising that we can’t enforce the presence of the deployment-scanner.
+* If the deployment-scanner is not in the config, the CLI operation will abort the plugin execution and log an error message.
+
+===== Workflow examples
+
+Description of the 2 workflows (with and without dev mode). These examples could be designed differently according to the content of the pom file.
+
+Nominal mode, full repackaging of the server done for each rebuild, server restarted after each rebuild (not viable):
+
+* mvn package ⇒ full repackaging
+* mvn wildfly-jar:run
+* (kill synchronous execution).
+* User make changes
+* mvn package ⇒ full repackaging
+* ...
+
+Dev mode, server built/started once, app automatically re-deployed.
+
+* mvn wildfly-jar:dev ⇒ hollow server built and started
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically deployed.
+* User make changes
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically re-deployed.
+* ...
+* mvn wildfly-jar:shutdown
+* User is fine with his changes, he can do the server+app packaging: mvn package
+
+NB: This workflow doesn’t require support in IDE, it is 100% Maven. It could be optimized with IDE plugin (e.g.: netbeans plugin to track static 
+files and avoid to re-package in this case).
+
+
+==== Cloud context
+
+The plugin allows to generate a bootable JAR usable in various Cloud execution contexts: custom container, Google JIB, Java s2i build.
+
+As an example, s2i binary build of the https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/mp-config/README.md[microprofile-config example] is provided. 
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (e.g.: auth properties files).
+* Ability to generate a runtime Maven repository in order to resolve the server artifacts from Maven local cache. 
+Huge benefit in term of JAR size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + Maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-wildfly.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (e.g.: Thorntail yaml file).
+* Package a custom standalone XML file. Standalone XML file is generated during build by the Maven plugin and can't be replaced. 
```

Comment:
Is there a technical reason behind not being able to provide my own XML?

Reply:
The custom XML file could be out of sync with the provisioned server. For example, the bootable jar server not containing JBoss modules required for some extensions present in the custom XML file.

Reply:
Thank you!